### PR TITLE
Added .hashCode() method in Java Strings terms folder

### LINKED ISSUE #1584: [Term Entry] Java Strings: `.hashCode()` Method
### Reason/inspiration (optional)

Add a new term entry under the [Java: Strings](https://www.codecademy.com/resources/docs/java/strings) concept. This entry should include:

- An explanation of the method.

- A "Syntax" section with a ```pseudo code snippet showing the general syntax.

- An "Example" section with a ```java code snippet showing how it works.

A markdown file named "hashCode.md" should be added to a folder named "hashCode" and placed in the terms folder located under content/java/concepts/strings/terms

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by AshutoshVJTI**
### Description

- Added new entry for Java Strings: .getBytes()
- Closes #1584 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/strings/terms/hashCode/hashCode.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+```
+
+This outputs the following:
+
+```shell
+667975538
+```
+
+**Note:** The hash code value may vary for the same object when the program is executed multiple times.
```
Comment (SSwiniarski) [inline content/java/concepts/strings/terms/hashCode/hashCode.md:50]:
```suggestion
> **Note:** The hash code value may vary for the same object when the program is executed multiple times.
```

**Review by SSwiniarski — COMMENTED** [review]
@AshutoshVJTI I did a first review and left a comment.

**Comment by AshutoshVJTI** [comment]
@SSwiniarski  I have updated the file with suggested change. Thanks!

**Review by SSwiniarski — APPROVED** [review]
@AshutoshVJTI Looks good for a second review.

File: content/java/concepts/strings/terms/hashCode/hashCode.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Methods'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`hashCode()`** method returns an integer hash code value for the object on which it is invoked. The hash code for a String object is computed like this:
```
Comment (yangc95) [inline content/java/concepts/strings/terms/hashCode/hashCode.md:14]:
```suggestion
The **`.hashCode()`** method returns an integer hash code value for the object on which it is invoked. The hash code for a String object is computed like this:
```

File: content/java/concepts/strings/terms/hashCode/hashCode.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+```pseudo
+s[0]*31^(n-1) + s[1]*31^(n-2) + ... + s[n-1]
```
Comment (yangc95) [inline content/java/concepts/strings/terms/hashCode/hashCode.md:17]: Can we explain this in a sentence or two?

File: content/java/concepts/strings/terms/hashCode/hashCode.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+```
+
+## Syntax
+
+```pseudo
+int hashCodeValue = objectName.hashCode();
+```
+
+- `hashCodeValue` is an integer variable that stores the hash code value returned by the .hashCode() method.
+- `objectName` is the object for which we want to obtain the hash code value.
```
Comment (yangc95) [inline content/java/concepts/strings/terms/hashCode/hashCode.md:27]:
```suggestion
- `hashCodeValue` is an integer variable that stores the hash code value returned by the `.hashCode()` method.
- `objectName` is the object for which we want to obtain the hash code value.
```

Let's refrain from using pronouns like "we" and just the second bullet point accordingly

File: content/java/concepts/strings/terms/hashCode/hashCode.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+}
+```
+
+This outputs the following:
+
+```shell
+667975538
+```
+
+> **Note:** The hash code value may vary for the same object when the program is executed multiple times.
```
Comment (yangc95) [inline content/java/concepts/strings/terms/hashCode/hashCode.md:50]: Why is this? Can we include a sentence to explain this

**Review by yangc95 — COMMENTED** [review]
@AshutoshVJTI Thanks for this PR! I conducted a review to further improve this entry. Please let me know if you have any concerns and we can reassess after the comments have been addressed

**Comment by AshutoshVJTI** [comment]
@yangc95 Thanks for the suggestions! I have updated the file with those. Please have a look and let me know if there are any further changes required.

File: content/java/concepts/strings/terms/hashCode/hashCode.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  - 'paths/computer-science'
+---
+
+The **`.hashCode()`** method returns an integer hash code value for the object on which it is invoked. The hash code for a `String` object is computed as follows:
+
+```pseudo
+s[0]*31^(n-1) + s[1]*31^(n-2) + ... + s[n-1]
+```
+
+Where s[i] is the ith character of the string, n is the length of the string, and ^ indicates exponentiation.
```
Comment (yangc95) [inline content/java/concepts/strings/terms/hashCode/hashCode.md:20]:
```suggestion
`s[i]` is the `i`-th character of the string, `n` is the length of the string, and `^` indicates exponentiation.
```

**Comment by yangc95** [comment]
🥳 @AshutoshVJTI here's the live link: https://www.codecademy.com/resources/docs/java/strings/hashCode

Screenshot 2023-03-27 at 6 09 13 PM

**Pull request merged**